http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/BootstrapProvider.java

```diff
@@ -56,8 +61,21 @@ public Bootstrap createBootstrap(String host, int port) {
                 .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, nettyConfiguration.connectTimeoutMillis())
                 .option(ChannelOption.SO_KEEPALIVE, nettyConfiguration.tcpKeepAlive())
                 .remoteAddress(InetSocketAddress.createUnresolved(host, port));
+
+        if (nettyConfiguration.isNonBlockingResolver()) {
+            bootstrap.resolver(nonBlockingResolverGroup());
+        }
+
         sdkChannelOptions.channelOptions().forEach(bootstrap::option);
 
         return bootstrap;
     }
+
+    private AddressResolverGroup<InetSocketAddress> nonBlockingResolverGroup() {
+        DnsNameResolverBuilder builder = new DnsNameResolverBuilder()
+            .channelType(NioDatagramChannel.class)
```

Comment:
Should we use the event loop group channel factory instead of hard-coding to NioDatagramChannels?

i.e. `.channelFactory(sdkEventLoopGroup.channelFactory())`

Reply:
But the expected type is different, `.channelType` expects `Class<? extends DatagramChannel>`, and channelFactory returns 
`ChannelFactory<? extends Channel>`. Or do you suggest I should create a channelType() method in `sdkEventLoopGroup`.

Reply:
`channelType` just delegates to `channelFactory`, so I was asking if we could just use `channelFactory` directly from this code. Now I see that `channelFactory` takes a `ChannelFactory<? extends DatagramChannel>` instead of a `ChannelFactory<? extends Channel>`, so my suggestion still won't work.

The concern I'm trying to fix is that in `SocketChannelResolver` chooses KQueue or Oio socket channels in some circumstances, and I'd like to repeat that decision-making logic for the datagram channels. Should we rename `SocketChannelResolver` to `ChannelResolver` and add a `resolveDatagramChannelFactory` so that we use the right datagram channel implementation for the platform?

Reply:
Makes sense, I'll find a way to implement that.

Reply:
I pushed changes that include this solution.